### PR TITLE
feat: central location for ColumnDataTypes

### LINKED ISSUE #133: Discussion: Should we represent the data types as a table or as a choice (enum-like value)?
Currently, we store data types in the database table app_columndatatypes. We add new data types by adding new migrations. 
We could solve this in another way by using "choices" https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.choices

Pros and cons of using a "table" vs "choices"
1. **Pro for choices**. It is easy to update the data types, as we don't need a migration to do it
2. **Pro for choices**. The data types are defined a single place in the code and not in several potential migrations
3. **Con for choices**. We will lack data type information when we join tables in the database (when using raw sql and bypassing the Django ORM). If we create data type ids as strings, it would get some information about the individual types, but we would never be able to change these ids.
4. **Con for choices**. Foreign key constraints?

### PR DESCRIPTION
**Pull request by philter87**
## Description

This PR does two things
- It splits the models.py file into several files and puts them in the models folder. 
- The COLUMN_DATA_TYPES are applied after a 'migrate', and not as part of the migration. This will simplify changes and all types will be located at a single location (and not in scattered over several migration files)

## Related Issues

Closes #133

Related #56

## Testing

- [X] Yes
- [ ] No, not needed (give a reason below)
- [ ] No, I need help writing them

## Reviewer Focus

- The file column_data_type.py and apps.py are the most important files. 
- The "ready()"-method is triggered automatically by Django on application startup. I have added post_migration "signal", which is a way to execute some code after the migrations have been applied. 
- The migration file "0001_initial.py" is no longer adding ColumnDataType as this is handled in a different way
- Ignore the files in the folder "models" (except column_data_type.py)

## Checklist

For all PRs that are not general documentation

- [X] Tests accompany or reflect changes to the code
- [X] Tests ran and passed locally
- [X] Ran the linter and formatter
- [X] Build has passed locally
- [X] Relevant documentation has been updated

### CODE REVIEW

**Review by lwjohnst86 — APPROVED** [review]
Had a look over (@philter87 also went over it with me today) and added some small docstring edits and other formatting fixes. Looks great 🤩

File: app/models/column_data_type.py
Comment (signekb) [inline app/models/column_data_type.py:1]: Nice! Just a small comment: The first two datatypes have examples of usage in their description. Should the others have that as well?

File: app/models/column_data_type.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ),
+    ColumnDataType(
+        id=2,
+        display_name="Text",
+        description="A text field without a limit to how many characters can be stored",
+    ),
+    ColumnDataType(
+        id=3,
+        display_name="Yes/No",
+        description="Also known as Yes/No, Boolean, or Logical. This stores data as either 0/1 or True/False. If you need/want to store actual words, use the Text field",
```
Comment (signekb) [inline app/models/column_data_type.py:38]:
```suggestion
        description="Also known as Boolean, or Logical. This stores data as either 0/1 or True/False. If you need/want to store actual words, use the Text field",
```
"Yes/No also known as "Yes/No"

File: app/models/column_data_type.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from django.db import models
+
+
+class ColumnDataType(models.Model):
+    display_name = models.TextField()
+    description = models.TextField()
+
+
+"""
+COLUMN_DATA_TYPES contains all the "sprout" types.
```
Comment (signekb) [inline app/models/column_data_type.py:10]:
```suggestion
COLUMN_DATA_TYPES contains all the data types in Sprout.
```

File: app/models/column_data_type.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Changing the properties of the content can be risky. See the list below
```
Comment (signekb) [inline app/models/column_data_type.py:12]:
```suggestion
Be cautious of changing the properties of the content of COLUMN_DATA_TYPES. See the list below
```

File: app/models/column_metadata.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+from app.models.column_data_type import ColumnDataType
+from app.models.table_metadata import TableMetadata
+
+
+class ColumnMetadata(models.Model):
+    """Object to create table with types of metadata from columns in imported data.
+
+    Args:
+        models: Values to store for metadata.
```
Comment (signekb) [inline app/models/column_metadata.py:11]: Not sure I understand this sentence? :)
We haven't names the arg `models` above, right? But just specified the type?

**Review by signekb — APPROVED** [review]
Very nice. I deleted my `db.sqlite3` and used the `just update-migration` command. It worked perfectly 👌  
I really like that we have moved the column datatypes away from the migrations!

**Comment by signekb** [comment]
Ah, I approved and now it's merged. Sorry, about that @philter87

**Comment by philter87** [comment]
> Ah, I approved and now it's merged. Sorry, about that @philter87

Okay, I will create a new PR with you suggested changes

**Pull request merged**